sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -5260,6 +5260,15 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val READ_DATA_SOURCE_CACHE_IGNORE_OPTIONS =
+    buildConf("spark.sql.legacy.readDataSourceCacheIgnoreOptions")
+      .internal()
+      .doc("When set to true, reading from file source table caches the first query plan and " +
+        "ignores subsequent changes in query options, e.g. `delimiters` option for csv tables.")
```

Comment:
optional: not sure if the example adds much value.  We ignore any subsequent change to query option right?  I guess example can be more useful where we document the options itself.